ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,52 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void callJfrog(StringBuffer &localpath, const char *manifestFilename, IPropertyTree &item)
+{
+    const char *repo = item.queryProp("@repository");
+    if(repo)
+    {
+        StringBuffer downloadPath("/tmp/HPCCSystems/jfrog_downloads/");
+        Owned<IFile> dir = createIFile(downloadPath.str());
+        if(!dir->createDirectory())
+            throw makeStringExceptionV(0,"Error creating download directory: %s", downloadPath.str());
+        StringBuffer filename(localpath);
+        getFileNameOnly(filename, false);
+        downloadPath.append(filename);
+        DBGLOG("jfrog dl path: %s", downloadPath.str());
+        StringBuffer filepath;
+        StringBuffer machine;
+        if (repo[0] == '/' && repo[1] == '/')
+            splitUNCFilename(repo, &machine, &filepath, nullptr, nullptr);
+        else
+            splitFilename(repo, nullptr, &filepath, nullptr, nullptr);
+        StringBuffer jfrog_cmd("jf rt dl --flat --quiet ");
+        if (!machine.isEmpty())
+            jfrog_cmd.appendf("--url=http:%s/artifactory ", machine.str());
+        StringBuffer userValue;
+        StringBuffer keyValue;
+        const char *repoName = item.queryProp("@repoName");
+        if (repoName && getSecretValue(keyValue, "jfrog", repoName, "key", true) && getSecretValue(userValue, "jfrog", repoName, "user", true))
+            jfrog_cmd.appendf("--user=%s --password=%s ", userValue.trimRight().str(), keyValue.trimRight().str());
+        jfrog_cmd.appendf("\"%s%s\" %s", filepath.str(), filename.str(), downloadPath.str());
+        StringBuffer errOut;
+        StringBuffer jfrogOut;
+        if (system("export CI=true") != 0)
+            throw makeStringExceptionV(0, "Error exporting jfrog setting: export CI=true");
+        else if(runExternalCommand(errOut, jfrogOut, jfrog_cmd.str(), nullptr) != 0)
+            throw makeStringExceptionV(0, "Error loading resource from jfrog: \n\"%s\"\n %s", jfrogOut.str(), errOut.str());
+        const char *md5 = item.queryProp("@md5");
```

Comment:
This check would make sense for non-jfrog as well - so I would move out of this function.

Reply:
I removed this check because it already occurs outside of the JFrog call.